packages/lib-sourcify/src/Validation/SolidityMetadataContract.ts

```diff
@@ -261,11 +268,12 @@ export class SolidityMetadataContract {
     };
 
     if (!compilationTarget || Object.keys(compilationTarget).length != 1) {
-      throw new Error(
+      logDebug(
         `Can't create JsonInput from metadata: Invalid compilationTarget in metadata: ${Object.keys(
           this.metadata.settings.compilationTarget,
         ).join(',')}`,
       );
+      throw new ValidationError('invalid_compilation_target');
```

Comment:
Related to the comment above, here we must pass the `compilationTargets` to the user

Reply:
I fixed this by passing the compilationTargets to the error message. Users of lib-sourcify will see this error message.

However, when you would use the job lookup endpoint, you wouldn't see them. For that, we would need to store the compilationTargets in the database (maybe in `verification_jobs_ephemeral`). Do you think that makes sense?

Reply:
In `verification_jobs_ephemeral` we already store the error messages, right?

Reply:
No, we don't store the error message. We just store the code, and construct the message with the getter from the code

Reply:
Imo, it's better to store these additional fields in the database than storing the messages themselves. We could add a json column `error_payload`, which contains just the neeeded fields for reconstructing the error messages.

Reply:
let's also have an opinion by @kuzdogan  since this would also affect the API design. Do we need to update the GET verificationJob result?

Reply:
We could just include these fields inside the error message, then no API change would be needed.

But we could also return that `errorPayload` field as a json object in the API. This would make it also machine-readable.

Reply:
> Probably the latter is even better considering that we could use it in the UI.

That's exactly what I'm thinking of, if from the UI we need to show which files are invalid or the keccak it's better to have those information available separately from the error message

Reply:
I decided with Marco to implement my proposal. I named the new column `error_data`. Now, we have customized error messages dependent on parameters without storing the error messages themselves in the database.

@marcocastignoli you can have another look.